tvb/simulator/integrators.py

```diff
@@ -109,13 +119,24 @@ def scheme(self, X, dfun, coupling, local_coupling, stimulus):
         msg = "Integrator is a base class; please use a suitable subclass."
         raise NotImplementedError(msg)
 
+    def constrain_state(self, X):
+        for sv_ind, sv_bounds in \
+                zip(self.constraint_state_variable_indices,
```

Comment:
a SIMD friendly form will do better performance wise
```python
vi, vb = self.constraint_state_variable_indices, self.constraint_state_variable_bounds
X[vi] = X[vi] + vb[:, 0]*(X[vi] < vb[:, 0]) + vb[:, 1]*(X[vi] > vb[:, 1])
```
but a numba dfun will be ideal to avoid temporaries.

Reply:
I did something similar removing the temp. But, mind that for one sided boundaries, the other side contains a None value + I got an error for the different arrays sizes in the addition there, because x[vi] is of size space x modes... So, I thought it is getting more complicated...